trestle/common/list_utils.py

```diff
@@ -79,7 +79,33 @@ def delete_item_from_list(item_list: List[Any], value: Any, key: Callable[[Any],
     return item_list
 
 
+def get_item_from_list(
+    item_list: Optional[List[Any]], value: Any, key: Callable[[Any], Any], remove: bool = False
+) -> Any:
```

Comment:
Please use `TG`  instead of `Any` here. The problem with `Any` is that Python type checker will not validate the type at all, as _anything_ can be returned. If we instead use `TG` the types will be "linked" and type checker will work.

Consider a very simple identity function
```
def identity(a: Any) -> Any:
    return a
```

And call it 
```
a_number = identity(23)
print(a_number + "test") <--- This will not notify user of the type error.
```

However, if we use `TG` it will.

Reply:
Now using TG for the items and TG2 for the value and callable result